lib/parser/provenance.h

```diff
@@ -208,6 +211,9 @@ class CookedSource {
   bool IsValid(ProvenanceRange r) const { return allSources_.IsValid(r); }
 
   std::optional<ProvenanceRange> GetProvenanceRange(CharBlock) const;
+  std::optional<CharBlock> GetCharBlockFromLineAndColumns(int, int, int) const;
+  std::optional<std::tuple<int, int, int>> GetLineAndColumnsFromCooked(
```

Comment:
Same thing here.

Reply:
I replaced the tuple with std::pair of the SourcePosition struct. Still verbose, but hopefully is much more clear what it represents.